pulsar-io/kafka-connect-adaptor/src/main/java/org/apache/pulsar/io/kafka/connect/KafkaConnectSource.java

```diff
@@ -135,6 +141,7 @@ private void initTransforms(Map<String, Object> config) {
                         ));
                     log.info("transform config: {}", transformConfig);
                     String predicateName = (String) transformConfig.get("predicate");
+                    boolean negated = (boolean) transformConfig.getOrDefault("negated", false);
```

Comment:
Just wondering if it would be useful to support cases where the value type of the property would be a `String` at runtime.

Perhaps something like this:
```suggestion
                    boolean negated = Boolean.parseBoolean(String.valueOf(transformConfig.getOrDefault("negated", "false")));
```

Not beautiful code, but would address all cases.